tests/filecheck/xdsl_mpi_pipeline_c.mlir

```diff
@@ -1,4 +1,4 @@
-// RUN: xdsl-opt -p "distribute-stencil{strategy=3d-grid slices=2,1,1 restrict_domain=false},canonicalize-dmp,convert-stencil-to-ll-mlir" %s | filecheck %s
+// RUN: xdsl-opt -p "canonicalize,cse,distribute-stencil{strategy=3d-grid slices=2,1,1 restrict_domain=false},shape-inference,canonicalize-dmp,stencil-bufferize,dmp-to-mpi{mpi_init=false},convert-stencil-to-ll-mlir,canonicalize,cse" %s | filecheck %s
```

Comment:
This test is doing more than the original one; I think it is stronger, and this is good.
Is there a reason behind that?

Reply:
IIRC, canonicalization gets rid of some useless stencil generated operations, that make my bufferization ungrecefully fail.
It could arguably be handled differently but figured might as well add those generic optimizations everywhere; they help with the code and just remove work for heavier optimization passes!

Reply:
Thanks for the reply, can you add this (your reply's useful info) as a comment/info/doc in some place you consider it appropriate?